src/mscorlib/shared/System/HashCode.cs

```diff
@@ -0,0 +1,433 @@
+            // methods.
+            
+            var val = (uint)value;
+            uint previousLength = _length++;
+            uint position = previousLength % 4;
+
+            // Switch can't be inlined.
+
+            if (position == 0)
+                _queue1 = val;
+            else if (position == 1)
+                _queue2 = val;
+            else if (position == 2)
+                _queue3 = val;
+            else // position == 3
+            {
+                if (previousLength == 3)
+                    Initialize(out _v1, out _v2, out _v3, out _v4);
+
+                _v1 = Round(_v1, _queue1);
+                _v2 = Round(_v2, _queue2);
+                _v3 = Round(_v3, _queue3);
+                _v4 = Round(_v4, val);
+            }
+        }
+
+        public int ToHashCode()
+        {
+            uint length = _length;
+            uint position = length % 4;
+
+            // If the length is less than 4, _v1 to _v4 don't contain anything
+            // yet. xxHash32 treats this differently.
+
+            uint hash;
+            if (length < 4)
+                hash = MixEmptyState();
+            else
+                hash = MixState(_v1, _v2, _v3, _v4);
+
+            // Multiply by 4 because we've been counting in bytes, not ints.
+
+            hash += length * 4;
+
+            // Mix what remains in the queue
+
+            // Switch can't be inlined right now, so emulate case statement
+            // fallthrough using goto.
+
+            // position refers to the *next* queue position in this method, so
+            // position == 1 means that _queue1 is populated; _queue2 would have
+            // been populated on the next call to Add.
+
+            if (position == 0) goto mixFinal;
+            hash = QueueRound(hash, _queue1);
+            if (position == 1) goto mixFinal;
+            hash = QueueRound(hash, _queue2);
+            if (position == 2) goto mixFinal;
+            hash = QueueRound(hash, _queue3);
+            
+            mixFinal:
+            hash = MixFinal(hash);
+            return (int)hash;
+        }
+
+#pragma warning disable 0809
+        // Obsolete member 'memberA' overrides non-obsolete member 'memberB'. 
+        // Disallowing GetHashCode and Equals is by design
+
+        // * We decided to not override GetHashCode() to produce the hash code 
+        //   as this would be weird, both naming-wise as well as from a
+        //   behavioral standpoint (GetHashCode() should return the object's
+        //   hash code, not the one being computed).
+
+        // * Even though ToHashCode() can be called safely multiple times on
+        //   this implementation, it is not part of the contract. If the
+        //   implementation has to change in the future we don't want to worry
+        //   about people who might have incorrectly used this type.
+
+        [Obsolete("HashCode is a mutable struct and should not be compared with other HashCodes.", error: true)]
```

Comment:
The full error message should be `"HashCode is a mutable struct and should not be compared with other HashCodes. Use ToHashCode to retrieve the computed hash code."`

Reply:
For the exception message too? Or just the obsolete message?

Reply:
I think some developers are going to mistakenly try to call `GetHashCode` instead of `ToHashCode`. It'd be worth including `"Use ToHashCode to retrieve the computed hash code."` in both the obsolete and exception message for `GetHashCode` as a hint of what to do.

Reply:
I would suggest making leaving off the `Use ToHashCode ...` for the Equals obsolete message and its exception message.